### PR TITLE
CONC-553: Fix SIGBUS when calling mysql_real_connect_start on ARM64 macOS

### PR DESCRIPTION
**Pull request by kotori2**
From [this stack overflow question](https://stackoverflow.com/questions/23852522/swapcontext-segfaults-when-swapping-to-a-ucontext-ts-uc-link)
> Evidently, that macro changes the layout of the relevant structures to match what those functions expect and require.

So I moved the `_XOPEN_SOURCE` macro to `CMakefile.txt`.

### CODE REVIEW

File: libmariadb/CMakeLists.txt [hunk 1]
```diff
@@ -4,6 +4,9 @@ INCLUDE_DIRECTORIES(${CC_SOURCE_DIR}/include
 ADD_DEFINITIONS(-D HAVE_COMPRESS)
 ADD_DEFINITIONS(-D LIBMARIADB)
 ADD_DEFINITIONS(-D THREAD)
+IF(APPLE)
+  ADD_DEFINITIONS(-D _XOPEN_SOURCE)
```
Comment (vuvova) [inline libmariadb/CMakeLists.txt:8]: shouldn't it be `_XOPEN_SOURCE=600` ?

Comment (kotori2) [reply libmariadb/CMakeLists.txt:8]: I have no idea why it was `600` as I checked `ucontext.h`: 
```
#ifdef _XOPEN_SOURCE
...
#else /* !_XOPEN_SOURCE */
#error The deprecated ucontext routines require _XOPEN_SOURCE to be defined
#endif /* _XOPEN_SOURCE */
```

Also it seems it was producing compilation errors if I set `_XOPEN_SOURCE` to `600`. Setting `-D _XOPEN_SOURCE` just works fine.

Comment (vuvova) [reply libmariadb/CMakeLists.txt:8]: ok, fair enough

**Comment by kotori2** [comment]
CI seems passing now.

**Comment by 0rangeFox** [comment]
@vuvova This is really necessary for M1 working when creating MySQL Client through code, and I confirm it's working perfectly with this simple changes.

**Comment by 9EOR9** [comment]
Manually merged into 3.1 branch

**Pull request closed without merging**